apps/fz_http/lib/fz_http_web/auth/json/authentication.ex

```diff
@@ -0,0 +1,28 @@
+defmodule FzHttpWeb.Auth.JSON.Authentication do
+  @moduledoc """
+  API Authentication implementation module for Guardian.
+  """
+  use Guardian, otp_app: :fz_http
+
+  alias FzHttp.{
+    ApiTokens,
+    Users
+  }
+
+  @impl Guardian
+  def subject_for_token(user, _claims) do
+    {:ok, user.id}
+  end
+
+  @impl Guardian
+  def resource_from_claims(%{"jti" => api_token_id}) do
+    with {:ok, api_token} <- ApiTokens.get_api_token(api_token_id),
+         false <- ApiTokens.expired?(api_token),
```

Comment:
By using this function you kinda leak the context internals, instead, we might just do `ApiTokens.get_unexpired_api_token(api_token_id)` which will have a condition in a query and return an error if it's expired.

Reply:
Nice catch, much cleaner!